### PR TITLE
Add react to rails app

### PR DESCRIPTION
**Pull request by Zel-hub7**
1. Installed the react dependencies.
2. Created migration files and added some random greetings.
3. Created api.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,10 +4,22 @@
   "dependencies": {
     "@hotwired/stimulus": "^3.2.2",
     "@hotwired/turbo-rails": "^8.0.0-beta.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "webpack": "^5.89.0",
```
Comment (mckent05) [inline package.json:9]: - You have done well so far, but please include the `react-router-dom` and `axios` libraries in your project dependencies, you are using them without installing them in your project. Please fix this 🙏

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    useEffect(()=>{
+        getData();
+    },[]);
+
+    return(
+        <h1>Greeting</h1>
+    )
+};
+
+export default Greeting;
```
Comment (mckent05) [inline app/javascript/components/Greeting.js:19]: - You have done well so far, but according to the project requirements, you are required to use a store, an action , and a reducer that uses the URL to the endpoint from the Rails API. All these should be implemented using Redux. You should use the[ createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) and [createSlice](https://redux-toolkit.js.org/api/createslice) methods in the redux toolkit. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Zel-hub7, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job setting up your rails-react application in one app 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/application.js [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
 import "./controllers"
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from "./components/App";
+import { BrowserRouter } from "react-router-dom";
+import { Provider } from "react-redux";
+
+ReactDOM.render(
+    <BrowserRouter>
+      <Provider store={store}>
```
Comment (Omar-Muhamad) [inline app/javascript/application.js:11]: - [ ] After installing gems and dependencies, setting up the database, and precompiling the assets, I have a blank screen and an error was raised here when I ran the dev server. Please fix that.

image

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Zel-hub7,

#### Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by shahadat3669 — APPROVED** [review]
Hi @Zel-hub7, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Store defined perfectly
✅ Show random message
✅ API working fine
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- [Optional] The branch name you have chosen does not provide a clear indication of the feature or action being implemented. It is important to use descriptive branch names that accurately describe the purpose of the branch. For example, instead of a generic name like "feature," consider using names like "**configure rails with react**" or "**combine react and rails**" that convey the specific feature or action being worked on. :pray: Please rename the branch to a meaningful name. Check out this [official document](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) for further help. :+1:
Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**